tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```
@@ -0,0 +1,1078 @@
+     * @return true if type1 is an instance of type2, false otherwise
+     */
+    public static boolean isInstanceOf(String type1, String type2, OracleDatapoint oracleDatapoint) {
+        return isInstanceOf(type1, type2, oracleDatapoint, true);
+    }
+
+    /**
+     * Compared to {@link #isInstanceOf(String, String, OracleDatapoint)}, this method returns
+     * true if type1 and type2 can be compared using the instanceof operator. To better understand this
+     * difference, consider the following use cases:
+     * <ul>
+     *     <li>{@code isInstanceOf("String", "Object", null)} returns {@code true}</li>
+     *     <li>{@code isInstanceOf("Object", "String", null)} returns {@code false}</li>
+     *     <li>{@code doesInstanceofCompile("String", "Object", null)} returns {@code true}</li>
+     *     <li>{@code doesInstanceofCompile("Object", "String", null)} returns {@code true}</li>
+     * </ul>
+     * In other words, this method returns true if the expression "var1 instanceof type2" would compile,
+     * where var1 is a variable of type1.
+     */
+    public static boolean doesInstanceofCompile(String type1, String type2, OracleDatapoint oracleDatapoint) {
+        ResolvedType resolvedType2 = tryToGetResolvedType(type2);
+        if (resolvedType2 == null) {
+            return false; // Either type2 is generic, or an unknown class. In both cases, type1 cannot be instanceof it
+        }
+        try {
+            List<String> type2TypeParameters = resolvedType2.asReferenceType().getTypeDeclaration().get().getTypeParameters()
+                    .stream().map(ResolvedTypeParameterDeclaration::getName).collect(Collectors.toList());
+            if (type2TypeParameters.contains(type1)) {
+                return true; // Special case: type1 is a generic and a type parameter of type2
+            }
+        } catch (UnsupportedOperationException|NoSuchElementException|NullPointerException ignored) {}
```

Document (as a comment within the empy block) why it is OK to ignore these exceptions.  Please do this for all empty catch blocks.

---

@AML14